HiP-Achievements/Controllers/ActionBaseController.cs

```diff
@@ -43,15 +40,15 @@ public ActionBaseController(EventStoreService eventStore, InMemoryCache cache)
             if (User.Identity.GetUserIdentity() == null)
                 return Forbid();
 
-            var validationResult = await ValidateActionArgs(args);
-            if (!validationResult.Success)
-                return validationResult.ActionResult;
-
-            var id = _entityIndex.NextId(ResourceTypes.Action);
-            await EntityManager.CreateEntityAsync(_eventStore, args, ResourceType, id, User.Identity.GetUserIdentity());
-            return Created($"{Request.Scheme}://{Request.Host}/api/Action/{id}", id);
+            try
+            {
+                var id = await _userStoreService.ExhibitVisitedAction.PostAsync(new ExhibitVisitedActionArgs() { EntityId = args.EntityId });
```

Comment:
This method is not only meant for ExhibitVisited actions. It should be used for all action types. So a better way would be to override this method in the child class instead of changing it in the base class